ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/ControlWrapper.java

```diff
@@ -270,7 +270,7 @@ public final void commit (boolean report_heuristics) throws Unavailable,
 		    // check termination status
 		    
 		    if (_controlImpl.getFinalStatus() != org.omg.CosTransactions.Status.StatusCommitted)
-                        throw new TRANSACTION_ROLLEDBACK();
+                        return;
```

Comment:
You are right - this is incorrect if we want to be consistent with ArjunaTransactionImple code which throws TRANSACTION_ROLLEDBACK in it's commit method - fixing.